### PR TITLE
chore: Split Fetch code

### PR DESCRIPTION
**Pull request by kibertoad**
## Rationale

I've started looking into https://twitter.com/bengl/status/1715958172400210065 to identify quick perf wins for the hot path, but quickly hit a roadblock, due to "monolithic block of code" style of code, that is hard to penetrate for an outsider. Since inline function declarations in the original code served as a way to pass closure variables, I've reimplemented this part of the code as a class, which exists precisely for carrying this kind of a context.

## Changes

This is pretty much the original code, just broken down into more manageable pieces.

### Features

N/A

### Bug Fixes

N/A

### Breaking Changes and Deprecations

Shouldn't be a breaking change

## Status

- [X] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [ ] Tested
- [ ] Benchmarked (**optional**)
- [ ] Documented
- [ ] Review ready
- [ ] In review
- [ ] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Comment by kibertoad** [comment]
@mcollina @KhafraDev Could you please take a look if this is something that makes sense within the undici codebase? Tests are still failing, but I don't want to invest much effort into something that will be rejected anyway.

File: lib/fetch/fetch-context/httpFetches.js [hunk 1]
```diff
@@ -0,0 +1,1673 @@
...
+      return await httpFetch(fetchParams)
+        .catch((err) => makeNetworkError(err))
+    }
+    default: {
+      return makeNetworkError('unknown scheme')
+    }
+  }
+}
+
+module.exports = {
```
Comment (kibertoad) [inline lib/fetch/fetch-context/httpFetches.js:1669]: I tried to split this into three separate files initially, but unfortunately that doesn't fly; as currently implemented, there is a circular reference across these three functions, so they can't be split at all. For me that sounds like a code smell, but this is a larger problem than what could be solved in this PR.

**Comment by KhafraDev** [comment]
If you have the fetch spec open, having the entirety of fetch in its own file makes maintaining it much easier. I'd also like to mention that some people have tried going through fetch for "easy" perf gains and there probably aren't that many that would be accepted.

**Comment by kibertoad** [comment]
fair point, I'll just close this then. This PR is making me feel miserable anyway :-D

**Pull request closed without merging**